portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting comments. */
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  private static final String COMMENT_ENTITY_KEY = "Comment";
+  private static final String COMMENT_PROPERTY_NAME = "userComment";
+  private static final String TIME_PROPERTY_NAME = "dateTimeCreated";
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(COMMENT_ENTITY_KEY);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+
+      Key commentEntityKey = KeyFactory.createKey(COMMENT_ENTITY_KEY, id);
+      datastore.delete(commentEntityKey);
+    }
+
+    response.setContentType("text/html;");
+    response.getWriter().println("");
+  }
+}
```

Comment:
Command line tools like `cat`, `grep`, etc. typically expect files to end with a newline character and can misbehave otherwise. Most editors have a configuration automatically ensure this, and I would recommend that you enable that option.

Reply:
+1